documentation/create_overview_doc/README.md

```diff
@@ -51,21 +51,27 @@ sh
 kt
 ```
 
+For flow map generation with Code2Flow, the tool currently supports the following languages, each with specific dependencies:
+
+- JavaScript: Requires Acorn for parsing. Ensure Acorn is installed and accessible in your environment.
+- Ruby: Uses Parser for parsing. Ensure the Parser gem is installed and accessible.
+- PHP: Employs PHP-Parser for parsing. PHP-Parser should be installed and accessible in your environment.
+- Python: No additional dependencies are required for Python projects.
+
+If your project uses a language not supported by Code2Flow for flow map generation, the documentation generation capabilities will still be available, but without the flow map visualization for those specific languages.
```

Comment:
**Optimization Issue**: The addition of language-specific dependencies for Code2Flow introduces a potential optimization issue. If the environment setup for each language is not efficiently managed, it could lead to redundant installations and increased setup time for the documentation generation tool. <br> **Fix**: Implement a dependency check mechanism that verifies if the required language parsers (e.g., Acorn for JavaScript, Parser gem for Ruby) are already installed before attempting to install them. This can prevent unnecessary installations and ensure that the setup process is as efficient as possible. <br> **Code Suggestion**: 
 ```
 +For flow map generation with Code2Flow, ensure the following dependencies are verified for installation only if not present:
 +- JavaScript: Check for Acorn.
 +- Ruby: Check for Parser gem.
 +- PHP: Check for PHP-Parser.
 ```